tests/manage/test_caching_config.py

```diff
@@ -11,74 +11,103 @@
     a specific dictionary. This is done by creating a temporary
     caching configuration file.
     """
+    config = create_config_instance()
 
     @contextlib.contextmanager
     def inner(config_dict):
-        with tempfile.NamedTemporaryFile() as handle:
-            yaml.dump({get_profile().name: config_dict}, handle, encoding='utf-8')
-            configure(config_file=handle.name)
+        for key, value in config_dict.items():
+            config.set_option(f'caching.{key}', value)
         yield
         # reset the configuration
-        configure()
+        for key in config_dict.keys():
+            config.unset_option(f'caching.{key}')
 
     return inner
 
 
-@pytest.fixture
-def use_default_configuration(configure_caching):  # pylint: disable=redefined-outer-name
-    """
-    Fixture to load a default caching configuration.
-    """
-    with configure_caching(
-        config_dict={
-            'default': True,
-            'enabled': ['aiida.calculations:arithmetic.add'],
-            'disabled': ['aiida.calculations:templatereplacer']
-        }
-    ):
-        yield
+def test_merge_deprecated_yaml(tmp_path):
+    """Test that an existing 'cache_config.yml' is correctly merged into the main config.
 
-
-def test_empty_enabled_disabled(configure_caching):
-    """Test that `aiida.manage.caching.configure` does not except when either `enabled` or `disabled` is `None`.
-
-    This will happen when the configuration file specifies either one of the keys but no actual values, e.g.::
-
-        profile_name:
-            default: False
-            enabled:
-
-    In this case, the dictionary parsed by yaml will contain `None` for the `enabled` key.
-    Now this will be unlikely, but the same holds when all values are commented::
-
-        profile_name:
-            default: False
-            enabled:
-                # - aiida.calculations:templatereplacer
-
-    which is not unlikely to occurr in the wild.
+    An AiidaDeprecationWarning should also be raised.
     """
-    with configure_caching(config_dict={'default': True, 'enabled': None, 'disabled': None}):
-        # Check that `get_use_cache` also does not except, and works as expected
-        assert get_use_cache(identifier='aiida.calculations:templatereplacer')
+    from aiida.common.warnings import AiidaDeprecationWarning
+    from aiida.manage import configuration
+    from aiida.manage.configuration import settings, load_profile, reset_profile, get_config_option
+
+    # Store the current configuration instance and config directory path
+    current_config = configuration.CONFIG
+    current_config_path = current_config.dirpath
+    current_profile_name = configuration.PROFILE.name
+
+    try:
+        reset_profile()
+        configuration.CONFIG = None
+
+        # Create a temporary folder, set it as the current config directory path
+        settings.AIIDA_CONFIG_FOLDER = str(tmp_path)
+        config_dictionary = {
```

Comment:
can one reuse the test sample 5.json here?

Reply:
done in d3291674c064165e147765708ae9cad1d363a279